parser.go

```diff
@@ -107,18 +107,18 @@ func (p *Parser) ParseUnverified(tokenString string, claims Claims) (token *Toke
 		if strings.HasPrefix(strings.ToLower(tokenString), "bearer ") {
 			return token, parts, NewValidationError("tokenstring should not contain 'bearer '", ValidationErrorMalformed)
 		}
-		return token, parts, &ValidationError{Inner: err, Errors: ValidationErrorMalformed}
+		return token, parts, &ValidationError{Inner: fmt.Errorf("malformed token header: %w", err), Errors: ValidationErrorMalformed}
 	}
 	if err = json.Unmarshal(headerBytes, &token.Header); err != nil {
-		return token, parts, &ValidationError{Inner: err, Errors: ValidationErrorMalformed}
+		return token, parts, &ValidationError{Inner: fmt.Errorf("malformed token header: %w", err), Errors: ValidationErrorMalformed}
```

Comment:
I've been going back and forth on this one. The intention is good, but is it going to break existing users?

There is _a lot_ to be desired in terms of improving errors. But is this something we should tackle in the current `/v4` or punt to a breaking release `/v5`.

Reply:
This discussion comes back to the point that we should discuss a common roadmap. I wonder, whether doing this in GitHub discussions is enough or if it makes sense to have a short zoom / Teams call on this to discuss a some roadmap ideas among the maintainers. Probably tricky to find a good time slot for this.

Reply:
We can opt to use semver properly in v4 and treat minor but non-revisional changes as these via releases (unless we strictly want to make the v4 release family v3 compatible). Those are the 2 options I see right now

@oxisto I think we can do fine with gh discussions for a middle ground between a chat and a an issue, for something more fluent we could use discord or something alike. I agree that coordinating a time slot might be difficult for a live conversation as we have tight schedules and different timezones but a chat server comes second place in communication efficiency so we can still gain something

Reply:
Yeah, maybe a discord server would be a nice idea.

Reply:
> We can opt to use semver properly in v4 and treat minor but non-revisional changes as these via releases

Ye that's fair, since we are using modules we can bump minor versions and just make sure to jot that down in the release change log.

I'm not sure I have the bandwidth for in-person chats. Happy to continue contributing asynchronously, also there is nothing pressing afacis. 

FWIW there is already a Gophers slack, and can create a "jwt" channel (maybe not) or just ping whoever is on there.

Reply:
The golang slack would seem like a obvious choice - many thousands of go users there and I believe they allow project channels.